screen2.0/src/app/search/search-helpers.ts

```diff
@@ -266,37 +452,178 @@ export function constructURL(
+    + `&TF=${outputT_or_F(newFilterCriteria.TF)}`
 
-  const classificationFilters = `&CA=${outputT_or_F(urlParams.CA)}&CA_CTCF=${outputT_or_F(urlParams.CA_CTCF)}&CA_H3K4me3=${outputT_or_F(
-    urlParams.CA_H3K4me3
-  )}&CA_TF=${outputT_or_F(urlParams.CA_TF)}&dELS=${outputT_or_F(urlParams.dELS)}&pELS=${outputT_or_F(urlParams.pELS)}&PLS=${outputT_or_F(
-    urlParams.PLS
-  )}&TF=${outputT_or_F(urlParams.TF)}`
+  const conservationFilters =
+    `&prim_s=${newFilterCriteria.prim_s}`
+    + `&prim_e=${newFilterCriteria.prim_e}`
+    + `&mamm_s=${newFilterCriteria.mamm_s}`
+    + `&mamm_e=${newFilterCriteria.mamm_e}`
+    + `&vert_s=${newFilterCriteria.vert_s}`
+    + `&vert_e=${newFilterCriteria.vert_e}`
 
-  const conservationFilters = `&prim_s=${urlParams.PrimateStart}&prim_e=${urlParams.PrimateEnd}&mamm_s=${urlParams.MammalStart}&mamm_e=${urlParams.MammalEnd}&vert_s=${urlParams.VertebrateStart}&vert_e=${urlParams.VertebrateEnd}`
+  const linkedGenesFilter =
+    `&genesToFind=${newFilterCriteria.genesToFind.join(',')}`
+    + `&distancePC=${outputT_or_F(newFilterCriteria.distancePC)}`
+    + `&distanceAll=${outputT_or_F(newFilterCriteria.distanceAll)}`
+    + `&CTCF_ChIA_PET=${outputT_or_F(newFilterCriteria.CTCF_ChIA_PET)}`
+    + `&RNAPII_ChIA_PET=${outputT_or_F(newFilterCriteria.RNAPII_ChIA_PET)}`
 
-  const linkedGenesFilter = `&genesToFind=${urlParams.genesToFind.join(',')}&distancePC=${outputT_or_F(urlParams.distancePC)}&distanceAll=${outputT_or_F(urlParams.distanceAll)}&distanceFromcCRE=${urlParams.distanceFromcCRE}&CTCF_ChIA_PET=${outputT_or_F(urlParams.CTCF_ChIA_PET)}&RNAPII_ChIA_PET=${outputT_or_F(urlParams.RNAPII_ChIA_PET)}`
+  const accessionsAndPage =
+    `&accessions=${accessions}`
+    + `&page=${page}`
 
-  const accessionsAndPage = `&accession=${urlParams.Accessions}&page=${urlParams.Page}`
+  const url =
+    `${urlBasics}`
+    + `${biosampleFilters}`
+    + `${chromatinFilters}`
+    + `${classificationFilters}`
+    + `${conservationFilters}`
+    + `${newFilterCriteria.genesToFind.length > 0 ? linkedGenesFilter : ""}`
+    + `${accessionsAndPage}`
 
-  const url = `${urlBasics}${biosampleFilters}${chromatinFilters}${classificationFilters}${conservationFilters}${urlParams.genesToFind.length > 0 ? linkedGenesFilter : ""}${accessionsAndPage}`
   return url
+}
+
+export function constructMainQueryParamsFromURL(searchParams: { [key: string]: string | undefined }): MainQueryParams {
+  return (
+    {
+      coordinates: {
+        //If bed intersecting, set chr start end to null
+        assembly: searchParams.assembly === "GRCh38" || searchParams.assembly === "mm10" ?
+          searchParams.assembly : "GRCh38",
+        chromosome: (searchParams.intersect && checkTrueFalse(searchParams.intersect)) ?
+          null : searchParams.chromosome ?? "chr11",
+        start: (searchParams.intersect && checkTrueFalse(searchParams.intersect)) ?
+          null : searchParams.start ?
+            Number(searchParams.start) : 5205263,
```

Comment:
just FYI => +(searchParams.start) also converts variable to number

Reply:
Changed, good suggestion